tests/test_requests.py

```diff
@@ -28,6 +28,34 @@ async def app(scope: Scope, receive: Receive, send: Send) -> None:
     assert response.json() == {"method": "GET", "url": "https://example.org:123/"}
 
 
+def test_request_lazy_load_property(test_client_factory: TestClientFactory) -> None:
+    async def app(scope: Scope, receive: Receive, send: Send) -> None:
+        request = Request(scope, receive)
+        assert not hasattr(request, "_url")
+        assert not hasattr(request, "_query_params")
+        assert not hasattr(request, "_json")
+        # trigger lazy loading
+        _, _, _ = request.url, request.query_params, await request.json()
+        assert hasattr(request, "_url")
+        assert hasattr(request, "_query_params")
+        assert hasattr(request, "_json")
+        data = {
+            "url": str(request.url),
+            "query_params": dict(request.query_params),
+            "json": await request.json(),
+        }
+        response = JSONResponse(data)
+        await response(scope, receive, send)
+
+    client = test_client_factory(app)
+    response = client.post("/42?foo=bar", json={"baz": "qux"})
+    assert response.json() == {
+        "url": "http://testserver/42?foo=bar",
+        "query_params": {"foo": "bar"},
+        "json": {"baz": "qux"},
+    }
```

Comment:
Which lines are this trying to cover?

Reply:
https://github.com/encode/starlette/blob/master/starlette/requests.py#L96
https://github.com/encode/starlette/blob/master/starlette/requests.py#L125
https://github.com/encode/starlette/blob/master/starlette/requests.py#L240

Reply:
Or should we add `pragma: no branch` to these lines too?

Reply:
The thing is that this test is doing too much, maybe we can divide it a bit? Or maybe just the `no branch`... I don't want us to try to satisfy the coverage just because we want to satisfy it, but actually create meaningful tests (naming wise as well).

Reply:
> Or maybe just the `no branch`

No problem, let's go with this.